signed/verifiers.go

```diff
@@ -80,55 +81,34 @@ func (v RSAVerifier) Verify(key data.Key, sig []byte, msg []byte) error {
 		return ErrInvalid
 	}
 
-	if err = rsa.VerifyPKCS1v15(rsaPub, crypto.SHA256, digest[:], sig); err != nil {
+	opts := rsa.PSSOptions{SaltLength: sha256.Size, Hash: crypto.SHA256}
+	if err = rsa.VerifyPSS(rsaPub, crypto.SHA256, digest[:], sig, &opts); err != nil {
 		logrus.Infof("Failed verification: %s", err)
 		return ErrInvalid
 	}
 	return nil
 }
 
-type RSAPemVerifier struct{}
+// RSAPSSPEMVerifier checks RSASSA-PSS signatures, extracting the public key
+// from an X509 certificate.
+type RSAPSSX509Verifier struct{}
 
-func (v RSAPemVerifier) Verify(key data.Key, sig []byte, msg []byte) error {
+// Verify does the actual check.
+func (v RSAPSSX509Verifier) Verify(key data.Key, sig []byte, msg []byte) error {
 	digest := sha256.Sum256(msg)
 
 	k, _ := pem.Decode([]byte(key.Public()))
-	cert, err := x509.ParseCertificate(k.Bytes)
-	if err != nil {
-		logrus.Errorf("Failed to parse public key: %s\n", err.Error())
-		return ErrInvalid
-	}
-
-	rsaPub, ok := cert.PublicKey.(*rsa.PublicKey)
-	if !ok {
-		logrus.Infof("Value returned from ParsePKIXPublicKey was not an RSA public key")
-		return ErrInvalid
-	}
-
-	if err = rsa.VerifyPKCS1v15(rsaPub, crypto.SHA256, digest[:], sig); err != nil {
-		logrus.Errorf("Failed verification: %s", err.Error())
+	if k == nil {
+		logrus.Infof("Failed to decode PEM-encoded x509 certificate")
 		return ErrInvalid
 	}
-	return nil
-}
-
-// RSAPSSVerifier checks RSASSA-PSS signatures
-type RSAPSSVerifier struct{}
```

Comment:
This verifier and its Verify functions need to remain in their original format. They provide compatibility for the Python implementation of TUF.